### PR TITLE
Forward Inventory to FleetDB

### PR DESCRIPTION
**Pull request by Alva8756**

### CODE REVIEW

File: pkg/api/routes/components_test.go [hunk 1]
```diff
@@ -67,6 +79,30 @@ func getComponentsHandler(t *testing.T, comps *fleetdb.ServerComponentSlice, cod
...
+		if byt != nil {
+			_, err := w.Write(byt)
+			if err != nil {
+				t.Fatalf("writing http response: %s", err.Error())
+			}
+		}
+	}
+}
+
 func TestFetchServerComponents(t *testing.T) {
```
Comment (DoctorVin) [inline pkg/api/routes/components_test.go:106]: I think going to a compile-time list instead of a runtime one will make this a _lot_ simpler.

**Review by DoctorVin — COMMENTED** [review]
My biggest criticism is that checking the types at runtime on each inventory. I think that will be more trouble than it's worth. It's OK for now, but when Joel lands that `bmc-toolbox` PR, we should use that instead. Getting the `slug` from the `name` is just a matter of down-casing.

File: internal/fleetdb/fleetdb.go [hunk 2]
```diff
@@ -48,69 +60,14 @@ func (fc fleetDBClient) GetComponents(ctx context.Context, id uuid.UUID, params
...
 
-func (fc fleetDBClient) UpdateAttributes(ctx context.Context, server *fleetdb.Server, dev *types.InventoryDevice, log *zap.Logger) error {
-	return createUpdateServerAttributes(ctx, fc.client, server, dev, log)
-}
-
-// Functions below may be refactored in the near future.
-func createUpdateServerAttributes(ctx context.Context, c *fleetdb.Client, server *fleetdb.Server, dev *types.InventoryDevice, log *zap.Logger) error {
-	newVendorData, newVendorAttrs, err := deviceVendorAttributes(dev)
+func (fc fleetDBClient) UpdateServerInventory(ctx context.Context, server *fleetdb.Server, dev *types.InventoryDevice, _ *zap.Logger, inband bool) error {
+	typeServer, err := fc.inventoryConverterInstance.ToRivetsServer(server.UUID.String(), server.FacilityCode, dev.Inv, *dev.BiosCfg)
```
[the remark this answers is not in the document]
Comment (Alva8756) [reply internal/fleetdb/fleetdb.go:64]: Sure, rename it.

File: pkg/api/routes/components_test.go (hunk 1 above)
Comment (Alva8756) [reply pkg/api/routes/components_test.go:106]: Hi Vince,
Can you please help provide more details about it? I am not sure what I need to do here.
By the way, I have verified the latest version of this PR in sandbox(`alloy` -> `CIS` -> `fleetdb`)

**Pull request merged**